sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -285,6 +286,116 @@ case class ArrayTransform(
   override def prettyName: String = "transform"
 }
 
+/**
+ * Sorts elements in an array using a comparator function.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(expr, func) - Sorts the input array in ascending order. The elements of the " +
+    "input array must be orderable. Null elements will be placed at the end of the returned " +
+    "array. Since 3.0.0 also sorts and returns the array based on the given " +
+    "comparator function. The comparator will take two nullable arguments " +
+    "representing two nullable elements of the array." +
+    "It returns -1, 0, or 1 as the first nullable element is less than, equal to, or greater " +
+    "than the second nullable element. If the comparator function returns other " +
+    "values (including NULL), the query will fail and raise an error.",
```

Comment:
We need to use multi-line string literal here.

```
/home/runner/work/spark/spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala:299: error: annotation argument needs to be a constant; found: "_FUNC_(expr, func) - Sorts the input array in ascending order. The elements of the ".+("input array must be orderable. Null elements will be placed at the end of the returned ").+("array. Since 3.0.0 also sorts and returns the array based on the given ").+("comparator function. The comparator will take two nullable arguments ").+("representing two nullable elements of the array.").+("It returns -1, 0, or 1 as the first nullable element is less than, equal to, or greater ").+("than the second nullable element. If the comparator function returns other ").+("values (including NULL), the query will fail and raise an error.")
    "than the second nullable element. If the comparator function returns other " +
                                                                                  ^
```

E.g., see `ArrayAggregate`.